### PR TITLE
test: Add ParserTest base class for standard positive tests

### PR DESCRIPTION
**Pull request by nathan-stender**
This PR adds a base test class for testing allotropy parsers.

By inheriting from this class and specifying a vendor, a `to_allotrope_test` will test all files in the corresponding `testdata` directory that do not contain `error` or `exclude` in their filepaths

This reduces boilerplate for creating a new allotropy parser, while also reducing the change of error (e.g. accidentally forgetting to add a test case to the test parameterization)

### CODE REVIEW

File: src/allotropy/testing/utils.py [hunk 1]
```diff
@@ -140,7 +144,7 @@ def _write_actual_to_expected(allotrope_dict: DictType, expected_file: str) -> N
 
 def validate_contents(
     allotrope_dict: DictType,
-    expected_file: str,
+    expected_file: str | Path,
```
Comment (stephenworlow) [inline src/allotropy/testing/utils.py:147]: nittiest of all nits, but my brain expects the order to be "Path | str"

File: tests/to_allotrope_test.py [hunk 2]
```diff
@@ -16,3 +20,23 @@ def test_allotrope_from_file_not_found() -> None:
 def test_allotrope_model_from_file_not_found() -> None:
     with pytest.raises(AllotropeConversionError, match=EXPECTED_ERROR_MESSAGE):
         allotrope_model_from_file(INVALID_FILE_PATH, Vendor.AGILENT_GEN5)
+
+
+# A parser can inherit from this test to automatically test all positive test cases of converting from file.
+class ParserTest:
```
Comment (stephenworlow) [inline tests/to_allotrope_test.py:26]: 🔥

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    return all(keyword not in str(path).lower() for keyword in EXCLUDE_KEYWORDS)
+
+
+def get_test_cases(testdata_dir: Path) -> list[Path]:
+    return [path for path in testdata_dir.rglob("*") if _is_valid_testcase(path)]
+
+
+def pytest_generate_tests(metafunc: pytest.Metafunc) -> None:
+    # Only parametrize test_file_path if class variable VENDOR is defined, signifying a ParserTest
+    if "test_file_path" in metafunc.fixturenames and metafunc.cls.VENDOR:
```
Comment (stephenworlow) [inline tests/conftest.py:26]: 🪄

**Review by stephenworlow — APPROVED** [review]

File: src/allotropy/testing/utils.py (hunk 1 above)
Comment (nathan-stender) [reply src/allotropy/testing/utils.py:147]: Well I did it so now you have to re-approve

**Review by stephenworlow — APPROVED** [review]

**Pull request merged**